testutil/fuse/hook.go

```diff
@@ -0,0 +1,167 @@
+func (h EmptyHook) PostRename(oldPatgh string, newPath string) (hooked bool, err error) {
+	return false, nil
+}
+
+// PreOpen is called after the real `open()` method, it provides a default implement.
+func (h EmptyHook) PreOpen(path string, flags uint32) (hooked bool, err error) {
+	return false, nil
+}
+
+// PostOpen is called after the real `open()` method, it provides a default implement.
+func (h EmptyHook) PostOpen(realRetCode int32) (hooked bool, err error) {
+	return false, nil
+}
+
+// PreRead is called before real `read()` method, it provides a default implement.
+func (h EmptyHook) PreRead(path string, length int64, offset int64) (hooked bool, err error) {
+	return false, nil
+}
+
+// PostRead is called after real `read()` method, it provides a default implement.
+func (h EmptyHook) PostRead(realRetCode int32, realBuf []byte) (hooked bool, err error) {
+	return false, nil
+}
+
+// PreWrite is called before real `write()` method, it provides a default implement.
+func (h EmptyHook) PreWrite(path string, buf []byte, offset int64) (hooked bool, err error) {
+	return false, nil
+}
+
+// PostWrite is called after real `write()` method, it provides a default implement.
+func (h EmptyHook) PostWrite(realRetCode int32) (hooked bool, err error) {
+	return false, nil
+}
+
+// PreMkdir is called before real `mkdir()` method, it provides a default implement.
+func (h EmptyHook) PreMkdir(path string, mode uint32) (hooked bool, err error) {
+	return false, nil
+}
+
+// PostMkdir is called after real `mkdir()` method, it provides a default implement.
+func (h EmptyHook) PostMkdir(realRetCode int32) (hooked bool, err error) {
+	return false, nil
+}
+
+// PreRmdir is called before real `rmdir()` method, it provides a default implement.
+func (h EmptyHook) PreRmdir(path string) (hooked bool, err error) {
+	return false, nil
+}
+
+// PostRmdir is called after real `rmdir()` method, it provides a default implement.
+func (h EmptyHook) PostRmdir(realRetCode int32) (hooked bool, err error) {
+	return false, nil
+}
+
+// PreOpenDir is called before real `opendir()` method, it provides a default implement.
+func (h EmptyHook) PreOpenDir(path string) (hooked bool, err error) {
+	return false, nil
+}
+
+// PostOpenDir is called after real `opendir()` method, it provides a default implement.
+func (h EmptyHook) PostOpenDir(realRetCode int32) (hooked bool, err error) {
+	return false, nil
+}
+
+// PreFsync is called before real `fsync()` method, it provides a default implement.
+func (h EmptyHook) PreFsync(path string, flags uint32) (hooked bool, err error) {
+	return false, nil
+}
+
+// PostFsync is called after real `fsync()` method, it provides a default implement.
+func (h EmptyHook) PostFsync(realRetCode int32) (hooked bool, err error) {
+	return false, nil
+}
+
+// FailingRenameHook implements the hook interface and fails on renaming operations.
+type FailingRenameHook struct {
+	// Add the Empty hook so the that this struct implements the hook interface.
```

Comment:
to follow the golang comment  standards.
```suggestion
	// EmptyHook is embedded so that the FailingRenameHook fully implements the hook interface.
```

Reply:
Oh my god, i just copied your comments here, i am so sorry.
Done